fvm/src/syscalls/bind.rs

```diff
@@ -94,6 +94,38 @@ fn memory_and_data<'a, K: Kernel>(
     Ok((Memory::new(mem), data))
 }
 
+fn charge_exec_units_for_gas(caller: &mut Caller<InvocationData<impl Kernel>>) -> Result<(), Trap> {
+    let exec_units = caller
+        .data_mut()
+        .calculate_exec_units_for_gas()
+        .map_err(|_| Trap::new("failed to calculate exec_units"))?;
+    if exec_units.is_negative() {
+        caller.add_fuel(u64::try_from(exec_units.saturating_neg()).unwrap_or(0))?;
```

Comment:
Technically... we want to add infinite fuel in the `unwrap_or` case? That seems like a bad idea. I think we need to turn this into a fatal error somehow?

Reply:
I mean, it can't happen, but it likely means we wrapped somewhere.

Reply:
Idk, I'm pretty comfortable saying "do nothing" if something literally impossible like this happens.